receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -234,6 +236,60 @@ func (k *K8sAPIServer) getServiceMetrics(clusterName, timestampNs string) []pmet
 	return metrics
 }
 
+func (k *K8sAPIServer) getStatefulSetMetrics(clusterName, timestampNs string) []pmetric.Metrics {
+	var metrics []pmetric.Metrics
+	statefulSets := k.leaderElection.statefulSetClient.StatefulSetInfos()
+	for _, statefulSet := range statefulSets {
+		fields := map[string]interface{}{
+			ci.MetricName(ci.TypeClusterStatefulSet, ci.SpecReplicas):   statefulSet.Spec.Replicas,            // statefulset_spec_replicas
+			ci.MetricName(ci.TypeClusterStatefulSet, ci.StatusReplicas): statefulSet.Status.Replicas,          // statefulset_status_replicas
+			ci.StatusReplicasAvailable:                                  statefulSet.Status.AvailableReplicas, // status_replicas_available
+		}
+		attributes := map[string]string{
+			ci.ClusterNameKey: clusterName,
+			ci.MetricType:     ci.TypeClusterStatefulSet,
+			ci.Timestamp:      timestampNs,
+			ci.PodNameKey:     statefulSet.Name,
```

Comment:
that's a good callout. I like the idea of adding new label but am also curious if it's adding much value to have this redundant information in most cases except `StatefulSet` since they will be the same for other workload types.